experimental/cloudstrapper/README.md

```diff
@@ -0,0 +1,178 @@
+
+# Chapter 1
+
+## 0. Prerequisites
+   - Create private repo on github to host helm charts. 
+     This information would be used in the build.yaml and cluster.yaml files when building and deploying Magma respectively. 
+
+   - Identify two security keys in the Build/Gateway regions to be used for the following
+     - Bootkey: Used only by Cloudstrapper instance
+     - Hostkey: Used by all Gateway instances
+       Both values are specified in the defaults.yaml vars file and embedded in hosts.
+     To generate keys through a playbook, see section 1 below.
+
+   - Create inventory directory on localhost to save keys, secrets etc. This directory
+     will be referred to as WORK_DIR and used as dirInventory in commands.
+
+   - Gather following credentials and update secrets.yaml on local machine in WORK_DIR. 
+     Use format from $CODE_DIR/playbooks/roles/vars/secrets.yaml as base.
+     - AWS Access and Secret keys
+     - Github username and PAT
+     - Dockerhub username and password 
+
+   - Understand key directories
+     - CODE_DIR: Directory hosting Magma code, typically in the ~/code/magma/experimental/cloudstrapper folder
+     - VARS_DIR: Directory where all variables reside, typically in the CODE_DIR/playbooks/roles/vars folder
+     - WORK_DIR: Directory where all working copies reside, typically n the ~/magma-experimental
+       folder
+
+## 1. Run aws-essentials to setup all AWS related components as a stack
+
+  The aws-essentials playbook will:
+  - Create boot and host keys if required using the keyCreate tag. Default is to not create keys.
+  - Create security group on the default VPC
+  - Create default bucket for shared storage. Ensure bucket does not exist by checking defaults.yaml
+
+  - Command:
+    ```
+    ansible-playbook aws-prerequisites.yaml -e 'awsTargetRegion=<< AWS Region >>' -e "dirInventory=<directory>" [ --tags keyCreate ]
+    ```
+  - Example:
+    ```
+    ansible-playbook aws-prerequisites.yaml -e 'awsTargetRegion=us-east-1' -e "dirInventory=~/magma-experimental/files" 
+    ``` 
+  - Result: Created stackMantleEssentials with common security group, S3 storage
```

Comment:
The idea is to have an option where everything that the customer needs is created by the tool itself with minimal or no dependencies from outside. The Cloudstrapper instance needs a security group to be instantiated. Instead of requesting for a pre-existing security group, the module creates it's own security group (customizable) thereby keeping it minimal touch. I will add text to explain this.